history.py

```diff
@@ -0,0 +1,28 @@
+from catalog import models as cmod
+
+class LastFiveMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+        # One-time configuration and initialization.
+
+    def __call__(self, request):
+
```

Comment:
Just a reminder to add ''' DEF DESCRIPTION ''' for your code, and to avoid excessive line spacing. Looks good thought!

Reply:
oh good point, ill do that.